drivers/sense/stream/argus/main.c

```diff
@@ -0,0 +1,246 @@
+
+	if (sensor_get_decoder(sensor, &decoder) < 0) {
+		return NULL;
+	}
+	return decoder;
+}
+
+static void argus_publish(struct context *ctx, argus_results_t *payload)
+{
+	stamp_msg(&ctx->data.stamp, k_uptime_ticks());
+
+	ctx->data.status = payload->Status;
+
+	ctx->data.has_bin = true;
+	ctx->data.bin.range = Z_SHIFT_Q31_TO_F32(payload->Bin.Range, 9);
+	ctx->data.bin.amplitude = Z_SHIFT_Q31_TO_F32(payload->Bin.Amplitude, 12);
+	ctx->data.bin.signal_quality = payload->Bin.SignalQuality;
+
+	ctx->data.has_frame = true;
+	ctx->data.frame.integration_time = payload->Frame.IntegrationTime;
+	ctx->data.frame.px_en_mask = payload->Frame.PxEnMask;
+	ctx->data.frame.ch_en_mask = payload->Frame.ChEnMask;
+	ctx->data.frame.state = payload->Frame.State;
+	ctx->data.frame.analog_integration_depth = Z_SHIFT_Q31_TO_F32(payload->Frame.AnalogIntegrationDepth, 10);
+	ctx->data.frame.digital_integration_depth = payload->Frame.DigitalIntegrationDepth;
+	ctx->data.frame.output_power = Z_SHIFT_Q31_TO_F32(payload->Frame.OutputPower, 12);
+	ctx->data.frame.bias_current = payload->Frame.BiasCurrent;
+	ctx->data.frame.pixel_gain = payload->Frame.PixelGain;
+	ctx->data.frame.pll_offset = payload->Frame.PllOffset;
+	ctx->data.frame.pll_ctrl_cur = payload->Frame.PllCtrlCur;
+
+	if (IS_ENABLED(CONFIG_ZROS_SENSE_STREAM_ARGUS_VERBOSE)) {
+		ctx->data.pixel_count = 32;
+		for (size_t i = 0; i < 32; i++) {
+			ctx->data.pixel[i].range = Z_SHIFT_Q31_TO_F32(payload->Pixels[i].Range, 9);
+			ctx->data.pixel[i].phase = Z_SHIFT_Q31_TO_F32(payload->Pixels[i].Phase, 1);
+			ctx->data.pixel[i].amplitude =
+				Z_SHIFT_Q31_TO_F32(payload->Pixels[i].Amplitude, 12);
+			ctx->data.pixel[i].status = payload->Pixels[i].Status;
+			ctx->data.pixel[i].range_window = payload->Pixels[i].RangeWindow;
+			ctx->data.pixel[i].amplitude_raw =
+				Z_SHIFT_Q31_TO_F32(payload->Pixels[i].AmplitudeRaw, 12);
+			ctx->data.pixel[i].uncorrelated_noise = 
+				Z_SHIFT_Q31_TO_F32(payload->Pixels[i].UncorrelatedNoise, 12);
+			ctx->data.pixel[i].snr = Z_SHIFT_Q31_TO_F32(payload->Pixels[i].SNR, 12);
+		}
+
+		ctx->data.has_pixel_ref = true;
+		ctx->data.pixel_ref.range = Z_SHIFT_Q31_TO_F32(payload->Pixels[32].Range, 9);
+		ctx->data.pixel_ref.phase = Z_SHIFT_Q31_TO_F32(payload->Pixels[32].Phase, 1);
+		ctx->data.pixel_ref.amplitude =
+			Z_SHIFT_Q31_TO_F32(payload->Pixels[32].Amplitude, 12);
+		ctx->data.pixel_ref.status = payload->Pixels[32].Status;
+		ctx->data.pixel_ref.range_window = payload->Pixels[32].RangeWindow;
+		ctx->data.pixel_ref.amplitude_raw =
+			Z_SHIFT_Q31_TO_F32(payload->Pixels[32].AmplitudeRaw, 12);
+		ctx->data.pixel_ref.uncorrelated_noise = 
+			Z_SHIFT_Q31_TO_F32(payload->Pixels[32].UncorrelatedNoise, 12);
+		ctx->data.pixel_ref.snr = Z_SHIFT_Q31_TO_F32(payload->Pixels[32].SNR, 12);
+
+		ctx->data.has_auxiliary = true;
+		ctx->data.auxiliary.vdd = Z_SHIFT_Q31_TO_F32(payload->Auxiliary.VDD, 12);
+		ctx->data.auxiliary.temp = Z_SHIFT_Q31_TO_F32(payload->Auxiliary.TEMP, 11);
```

Comment:
I think some of these are sign and others unsigned Q values does the Z_SHIFT_Q31_TO_F32 allow for specifying if base Q was a signed or not encoding?